src/Microsoft.VisualStudio.Services.Agent/ConfigurationStore.cs

```diff
@@ -57,6 +57,19 @@ public sealed class AgentSettings
 
         [DataMember(EmitDefaultValue = false)]
         public string CollectionName { get; set; }
+
+        [DataMember(EmitDefaultValue = false)]
+        public string AutoLogonUserName { get; set; }
```

Comment:
you shouldn't put autologon setting into the core config store. follow the windows service pattern, create a .autologon file.